### PR TITLE
Visualize donating your money now vs saving it for later

### PR DESCRIPTION
**Pull request by lukasberglund**
This is what it looks like https://screenshot.googleplex.com/hyz7xesABr9

### CODE REVIEW

File: portfolio/src/main/webapp/js/donate-or-save.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// limitations under the License.
+
+const GROWTH_RATE = 1.07;
+
+/** 
+ *  Simulate how much money you would be donating over a career that lasts n years with 
+ *  certain annual income, IF YOU INVESTED IT AND DONATED LATER.
+ */
+export function simulateSave(annualDonation, numYears) {
+  let total = 0;
```
Comment (alygerharter) [inline portfolio/src/main/webapp/js/donate-or-save.js:22]: nit: should this be totalSaved?

Comment (lukasberglund) [reply portfolio/src/main/webapp/js/donate-or-save.js]: Sure

File: portfolio/src/main/webapp/js/donate-or-save.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const GROWTH_RATE = 1.07;
+
+/** 
+ *  Simulate how much money you would be donating over a career that lasts n years with 
+ *  certain annual income, IF YOU INVESTED IT AND DONATED LATER.
+ */
+export function simulateSave(annualDonation, numYears) {
```
Comment (prabenzom) [inline portfolio/src/main/webapp/js/donate-or-save.js:21]: nit: i'd argue you should add in some negative value to factor in that society isn't being helped in the present

File: portfolio/src/main/webapp/js/donate-or-save.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  let totalSaved = 0;
+  let snapshots = new Array();
+
+  for (let cycle = 0; cycle <= numYears; cycle++) {
+    snapshots.push(totalSaved);
+    totalSaved *= GROWTH_RATE;
+    totalSaved += annualDonation;
+    annualDonation *= GROWTH_RATE;
```
Comment (prabenzom) [inline portfolio/src/main/webapp/js/donate-or-save.js:29]: nit: it looks like this is a compound interest formula with annual contributions? http://www.moneychimp.com/articles/finworks/fmbasinv.htm

File: portfolio/src/main/webapp/js/donate-or-save.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+/** 
+ *  Simulate how much money you would be donating over a career that lasts n years with 
+ *  certain annual income. We assume that your income grows by the same amount that a 
+ *  stock portfolio grows over time.
+ */
+export function simulateDonate(annualDonation, numYears) {
+  let totalDonated = 0;
+  let snapshots = new Array();
+
+  for (let cycle = 0; cycle <= numYears; cycle++) {
```
Comment (prabenzom) [inline portfolio/src/main/webapp/js/donate-or-save.js:44]: nit: you could argue that donating now adds value to society in present time that will grow over the years

File: portfolio/src/main/webapp/js/donate-or-save.js (hunk 2 above)
Comment (alygerharter) [reply portfolio/src/main/webapp/js/donate-or-save.js:21]: hahaha valid point

Comment (lukasberglund) [reply portfolio/src/main/webapp/js/donate-or-save.js:21]: This is an interesting question. If I am interpreting this correctly you are arguing that we discount the future as opposed to the present, i.e we assign less value to improvements in the future vs improvements in the present. Economists do that routinely, but philosophers are kind of against it. [Here is a discussion of the issue.](http://reflectivedisequilibrium.blogspot.com/2012/05/philosophers-vs-economists-on.html)

Comment (lukasberglund) [reply portfolio/src/main/webapp/js/donate-or-save.js:21]: I could add an option where you can set a discount rate later on if that sounds fair to you?

File: portfolio/src/main/webapp/js/donate-or-save.js (hunk 3 above)
Comment (lukasberglund) [reply portfolio/src/main/webapp/js/donate-or-save.js:29]: It is. Do you agree with my use of compound interest?

File: portfolio/src/main/webapp/js/donate-or-save.js (hunk 4 above)
Comment (lukasberglund) [reply portfolio/src/main/webapp/js/donate-or-save.js:44]: That's very true. I could add a disclaimer about that on my website if you want.

**Pull request merged**